src/pages/generate.tsx

```diff
@@ -116,98 +134,362 @@ export default function ClientIdentifierGenerator() {
+      marginLeft="auto"
+      marginRight="auto"
+    >
+      <Grid container item>
+        <Grid container item justifyContent="center">
+          <Grid item>
+            <Typography variant="h4">
+              Generate a Client Identifier Document
+            </Typography>
+          </Grid>
+        </Grid>
+        <Grid container item padding={2} spacing={1} marginLeft={2}>
+          <Grid item>
+            <Typography variant="body1">
+              This generator helps you set up a Client Identifier Document. A
+              Client Identifier Document describes your application, the client,
+              to a Solid OpenID Connect Provider (Solid OIDC Provider).
+            </Typography>
+          </Grid>
+          <Grid item>
+            <Typography variant="body1">
+              Do you want your users to login to their solid pods? Then you need
+              a Client Identifier Document! The user&apos;s OIDC Authentication
+              Provider will check your Client Identifier Document, let the user
+              authenticate, and hand your application an OIDC ID token.
+            </Typography>
+          </Grid>
+          <Grid item>
+            <Typography variant="body1">
+              With the token, your application can request resources from the
+              user&apos;s pod.
+            </Typography>
+          </Grid>
+        </Grid>
+        <Grid
+          container
+          paddingLeft={2}
+          paddingRight={2}
+          direction="column"
+          spacing={1}
+        >
+          <Grid item>
+            <Typography variant="h5">
+              Your app&apos;s (client&apos;s) information
+            </Typography>
+          </Grid>
+          <Grid container item>
+            <Formik
+              onSubmit={onSubmit}
+              onReset={onReset}
+              initialValues={initialFormValues}
+            >
+              {(form) => (
+                <Form>
+                  <Grid
+                    container
+                    item
+                    direction="column"
+                    spacing={2}
+                    padding={1}
+                  >
+                    <Grid container item>
+                      <TextField
+                        name="clientId"
+                        inputProps={{ inputMode: "url" }}
+                        required
+                        value={form.values.clientId}
+                        onChange={form.handleChange}
+                        fullWidth
+                        size="small"
+                        label="Client Identifier URI"
+                        helperText="The URI where your
+                          Client Identifier Document is located. It identifies your
+                          application, the client, to the Solid OIDC Provider.
+                          The Client Identifier Document should be a static resource
+                          and publicly accessible.
```

Comment:
```suggestion
                          The Client Identifier Document should be publicly accessible.
```
This is again related to what we discussed regarding the query parameters, but there is no reason to prevent the client identifier from being a "dynamic" resource.